app/models/upload.rb

```diff
@@ -268,14 +268,14 @@ def local?
   def fix_dimensions!
     return if !FileHelper.is_supported_image?("image.#{extension}")
 
-    path =
-      if local?
-        Discourse.store.path_for(self)
-      else
-        Discourse.store.download(self).path
-      end
-
     begin
+      path =
+        if local?
+          Discourse.store.path_for(self)
+        else
+          Discourse.store.download!(self).path
+        end
```

Comment:
Moved inside the `begin` to piggyback off unexpected error handling that was already implemented for issues with the image processing steps.